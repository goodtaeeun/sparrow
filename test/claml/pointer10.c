struct strt {
  int Length;
};

int main() {
  struct strt *acts = (struct strt *)malloc(sizeof(struct strt));
  acts->Length = 3;

  struct strt *act_implicit = &*acts;
  act_implicit->Length = 4;

  struct strt temp = *acts;
  struct strt *act_explicit = &temp;
  act_explicit->Length = 5;

  sparrow_print(acts);
  sparrow_print(act_implicit);
  sparrow_print(act_explicit);
  sparrow_print(acts->Length);
  sparrow_print(act_implicit->Length);
  sparrow_print(act_explicit->Length);

  return 0;
}
