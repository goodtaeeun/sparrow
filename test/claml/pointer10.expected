/* Generated by CIL v. 1.7.4 */
/* print_CIL_Input is false */

#line 1 "pointer10.c"
struct strt {
  int Length;
};
#line 5 "pointer10.c"
int main(void) {
  struct strt *acts;
  int __cil_tmp2;
  struct strt *act_implicit;
  struct strt temp;
  struct strt *act_explicit;

  {
    {
#line 6
      __cil_tmp2 = malloc(sizeof(struct strt));
#line 6
      acts = (struct strt *)__cil_tmp2;
#line 7
      acts->Length = 3;
#line 9
      act_implicit = acts;
#line 10
      act_implicit->Length = 4;
#line 12
      temp = *acts;
#line 13
      act_explicit = &temp;
#line 14
      act_explicit->Length = 5;
#line 16
      sparrow_print(acts);
#line 17
      sparrow_print(act_implicit);
#line 18
      sparrow_print(act_explicit);
#line 19
      sparrow_print(acts->Length);
#line 20
      sparrow_print(act_implicit->Length);
#line 21
      sparrow_print(act_explicit->Length);
    }
#line 23
    return (0);
  }
}
